skimage/draw/draw3d.py

```diff
@@ -63,6 +64,178 @@ def ellipsoid(a, b, c, spacing=(1., 1., 1.), levelset=False):
+    intrinsic : bool optional (default True)
+        Specify if the rotations are intrinsic (``True``) or extrinsic
+        (``False``).
+
+        - *intrinsic*: rotations occur about the rotating coordinate system
+        - *extrinsic*: rotations occur about the original coordinate system
+    spacing : (3,) array-like of float, optional (default None)
+        Spacing in each spatial dimension. The order is (z, y, x). If
+        ``spacing`` is ``None``, ``1.`` is set to all dimensions.
+
+    Returns
+    -------
+    dd, rr, cc : ndarray of int
+        Voxel coordinates of ellipsoid.
+        May be used to directl index into an array, e.g.
+        ``img[dd, rr, cc] = 1``
+
+    Raises
+    ------
+    ValueError
+        If the length of ``center`` is not 3.
+    ValueError
+        If the length of ``axis_lengths`` is not 3.
+    ValueError
+        If the length of ``angles`` is not 3.
+    ValueError
+        If the length of ``spacing`` is not 3.
+    ValueError
+        If ``axes`` is invalid.
+
+    Notes
+    ------
+    ``angles`` are defined as the Euler angles based on ``axes`` and
+    ``intrinsic`` parameters. The following example shows the ``0-2-0``
+    (``z-x-z``) intrinsic rotations.
+
+    .. math::
+
+        R(\phi) = \begin{bmatrix}
+                      1 &        0 &         0 \\
+                      0 & \cos\phi & -\sin\phi \\
+                      0 & \sin\phi &  \cos\phi \\
+                  \end{bmatrix}
+
+        R(\theta) = \begin{bmatrix}
+                        \cos\theta & -\sin\theta & 0 \\
+                        \sin\theta &  \cos\theta & 0 \\
+                                 0 &           0 & 1 \\
+                    \end{bmatrix}
+
+        R(\psi) = \begin{bmatrix}
+                      1 &        0 &         0 \\
+                      0 & \cos\psi & -\sin\psi \\
+                      0 & \sin\psi &  \cos\psi \\
+                           \end{bmatrix}
+
+        \\
+
+        R(\phi, \theta, \psi) = (R(\psi)R(\theta)R(\phi))^\intercal
+
+    """
+
+    if len(center) != 3:
+        raise ValueError(f'len(center) should be 3 but got {len(center)}')
+    center = np.array(center)
+
+    if len(axis_lengths) != 3:
+        raise ValueError(
+            f'len(axis_lengths) should be 3 but got {len(axis_lengths)}')
+    axis_lengths = np.array(axis_lengths)
+
+    R = np.eye(3)
+    if angles is not None:
+        if len(angles) != 3:
+            raise ValueError(
+                'len(angles) should be 3 but got',
+                len(angles))
+        if axes is None:
+            axes = range(3)
+        if (len(axes) != 3
+            or not set(axes).issubset(range(3))
+            or axes[0] == axes[1]
+                or axes[1] == axes[2]):
+            raise ValueError(f'axes: {axes} is invalid')
+        if intrinsic:
+            for i, angle in zip(axes, angles):
+                R = _euler_rotation(i, angle) @ R
+            R = R.T
```

Comment:
Since $(R_2 R_1)^T = R_1^T R_2^T$, this:
```python
for i, angle in zip(axes, angles):
    R = _euler_rotation(i, angle) @ R
R = R.T
```
Should be equivalent to:
```python
for i, angle in zip(axes, angles):
    R = R @ _euler_rotation(i, angle).T
```
Which would be a bit more consistent with the case where `intrinsic == False`.

I imagine there's a good reason why these rotations are inverted/transposed?